src/agentlab/llm/chat_api.py

```diff
@@ -200,24 +182,41 @@ def make_model(self):
 class ChatModel(ABC):
 
     @abstractmethod
-    def __init__(self, model_name, api_key=None, temperature=0.5, max_tokens=100):
+    def __init__(self, model_name, api_key=None, temperature=0.5, max_tokens=100, max_retry=1):
         self.model_name = model_name
         self.temperature = temperature
         self.max_tokens = max_tokens
+        self.max_retry = max_retry
 
-        self.client = tracking.OpenAI()
+        self.client = OpenAI()
 
         self.input_cost = 0.0
         self.output_cost = 0.0
 
     def __call__(self, messages: list[dict]) -> dict:
```

Comment:
can we log in `self` the number of retries and perhaps categorize them (e.g.
rate limit, server error...).
and we implement get_stats() so the agent can put it in its own stats.